src/analyze/analyze.c

```diff
@@ -1212,7 +1247,63 @@ static int dot(int argc, char *argv[], void *userdata) {
+                        { "Style", "Requires",          config_parse_string,  0, &dependency_style[UNIT_REQUIRES]  },
+                        { "Style", "Requisite",         config_parse_string,  0, &dependency_style[UNIT_REQUISITE] },
+                        { "Style", "Wants",             config_parse_string,  0, &dependency_style[UNIT_WANTS]     },
+                        { "Style", "Conflicts",         config_parse_string,  0, &dependency_style[UNIT_CONFLICTS] },
+                        { "Style", "BindsTo",           config_parse_string,  0, &dependency_style[UNIT_BINDS_TO]  },
+                        { "Style", "PartOf",            config_parse_string,  0, &dependency_style[UNIT_PART_OF]   },
+                        { "Style", "OnFailure",         config_parse_string,  0, &dependency_style[UNIT_ON_FAILURE]},
+                        { "Style", "Triggers",          config_parse_string,  0, &dependency_style[UNIT_TRIGGERS]  },
+                        {}
+        };
+
+        dependency_color[UNIT_AFTER]      = strdup("green");
+        dependency_color[UNIT_REQUIRES]   = strdup("black");
+        dependency_color[UNIT_REQUISITE]  = strdup("blue");
+        dependency_color[UNIT_WANTS]      = strdup("grey66");
+        dependency_color[UNIT_CONFLICTS]  = strdup("red");
+        dependency_color[UNIT_BINDS_TO]   = strdup("magenta");
+        dependency_color[UNIT_PART_OF]    = strdup("black");
+        dependency_color[UNIT_ON_FAILURE] = strdup("orange");
+        dependency_color[UNIT_TRIGGERS]   = strdup("grey66");
+
+        dependency_style[UNIT_AFTER]      = strdup("solid");
+        dependency_style[UNIT_REQUIRES]   = strdup("solid");
+        dependency_style[UNIT_REQUISITE]  = strdup("solid");
+        dependency_style[UNIT_WANTS]      = strdup("solid");
+        dependency_style[UNIT_CONFLICTS]  = strdup("solid");
+        dependency_style[UNIT_BINDS_TO]   = strdup("solid");
+        dependency_style[UNIT_PART_OF]    = strdup("dashed");
+        dependency_style[UNIT_ON_FAILURE] = strdup("solid");
+        dependency_style[UNIT_TRIGGERS]   = strdup("dotted");
+
+        if (!dependency_color[UNIT_AFTER]      || !dependency_style[UNIT_AFTER]      ||
+            !dependency_color[UNIT_REQUIRES]   || !dependency_style[UNIT_REQUIRES]   ||
+            !dependency_color[UNIT_REQUISITE]  || !dependency_style[UNIT_REQUISITE]  ||
+            !dependency_color[UNIT_WANTS]      || !dependency_style[UNIT_WANTS]      ||
+            !dependency_color[UNIT_CONFLICTS]  || !dependency_style[UNIT_CONFLICTS]  ||
+            !dependency_color[UNIT_BINDS_TO]   || !dependency_style[UNIT_BINDS_TO]   ||
+            !dependency_color[UNIT_PART_OF]    || !dependency_style[UNIT_PART_OF]    ||
+            !dependency_color[UNIT_ON_FAILURE] || !dependency_style[UNIT_ON_FAILURE] ||
+            !dependency_color[UNIT_TRIGGERS]   || !dependency_style[UNIT_TRIGGERS]
+        )
```

Comment:
please implement this differently: use a static const array of strings with the colors, indexed by the dependency type, then initialize this with a loop

Reply:
done